micropub.php

```diff
@@ -810,31 +810,44 @@ private static function check_error( $result ) {
 			return $result;
 		}
 
+
+		public static function get_micropub_endpoint() {
+			return apply_filters( 'micropub_endpoint', site_url( '?micropub=endpoint' ) );
+		}
+
+		public static function get_authentication_endpoint() {
+			return apply_filters( 'micropub_authentication_endpoint', MICROPUB_AUTHENTICATION_ENDPOINT );
+		}
+
+		public static function get_token_endpoint() {
+			return apply_filters( 'micropub_endpoint', MICROPUB_TOKEN_ENDPOINT );
+		}
+
 		/**
 	 * The micropub autodicovery meta tags
 	 */
 		public static function html_header() {
-			echo '<link rel="micropub" href="' . site_url( '?micropub=endpoint' ) . '">';
-			echo '<link rel="authorization_endpoint" href="' . MICROPUB_AUTHENTICATION_ENDPOINT . '">';
-			echo '<link rel="token_endpoint" href="' . MICROPUB_TOKEN_ENDPOINT . '">';
+			printf( '<link rel="micropub" href="%s" />' . PHP_EOL, static::get_micropub_endpoint() );
```

Comment:
I don't have a strong opinion on either side of the / debate(it seems there are differing opinions on it. I liked the changes that @pfefferle made to similar entries in the webmention plugins because they are cleaner and addressed another issue, and he had the / .